### PR TITLE
edit page changes as per new page header

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
* Breadcrumb added with a name while editing a channel and recepient group page

### Issues Resolved
Image 9-6-24 at 8 19 AM
Image 9-6-24 at 8 19 AM

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/CreateChannel/CreateChannel.tsx [hunk 1]
```diff
@@ -144,15 +144,25 @@ export function CreateChannel(props: CreateChannelsProps) {
...
+      BREADCRUMBS.NOTIFICATIONS,
       BREADCRUMBS.CHANNELS,
-      props.edit ? BREADCRUMBS.EDIT_CHANNEL : BREADCRUMBS.CREATE_CHANNEL]);
-    window.scrollTo(0, 0);
+    ];
+    if (edit) {
+      if (getUseUpdatedUx()) {
+        breadcrumbs.push(BREADCRUMBS.EDIT_CHANNEL_DETAILS(name))
+      }
+      breadcrumbs.push(BREADCRUMBS.EDIT_CHANNEL)
```
Comment (amsiglan) [inline public/pages/CreateChannel/CreateChannel.tsx:156]: Do we need this if we are showing the name?

File: public/pages/Emails/CreateRecipientGroup.tsx [hunk 2]
```diff
@@ -66,19 +66,25 @@ export function CreateRecipientGroup(props: CreateRecipientGroupProps) {
...
-        ? BREADCRUMBS.EDIT_RECIPIENT_GROUP
-        : BREADCRUMBS.CREATE_RECIPIENT_GROUP,
-    ]);
-    window.scrollTo(0, 0);
+    ];
+    if (edit) {
+      if (getUseUpdatedUx()) {
+        breadcrumbs.push(BREADCRUMBS.EDIT_RECIPIENT_GROUP_DETAILS(name))
+      }
+      breadcrumbs.push(BREADCRUMBS.EDIT_RECIPIENT_GROUP)
```
Comment (amsiglan) [inline public/pages/Emails/CreateRecipientGroup.tsx:78]: Same question, do we need this edit breadcrumb?

File: public/pages/CreateChannel/CreateChannel.tsx [hunk 3]
```diff
@@ -144,15 +144,25 @@ export function CreateChannel(props: CreateChannelsProps) {
   });
 
   useEffect(() => {
-    setBreadcrumbs([ BREADCRUMBS.NOTIFICATIONS,
+    const { edit } = props;
+    const breadcrumbs = [
+      BREADCRUMBS.NOTIFICATIONS,
       BREADCRUMBS.CHANNELS,
-      props.edit ? BREADCRUMBS.EDIT_CHANNEL : BREADCRUMBS.CREATE_CHANNEL]);
-    window.scrollTo(0, 0);
```
[the remark this answers is not in the document]
Comment (riysaxen-amzn) [reply public/pages/CreateChannel/CreateChannel.tsx:150]: addressed

File: public/pages/Emails/CreateRecipientGroup.tsx (hunk 2 above)
Comment (riysaxen-amzn) [reply public/pages/Emails/CreateRecipientGroup.tsx:78]: yes, we do other wise just the name breadcrumb doesn't appear, it comes like this if i remove the line `breadcrumbs.push(BREADCRUMBS.EDIT_RECIPIENT_GROUP)`
Image 9-6-24 at 10 43 AM

File: public/pages/CreateChannel/CreateChannel.tsx (hunk 1 above)
Comment (riysaxen-amzn) [reply public/pages/CreateChannel/CreateChannel.tsx:156]: same comment as https://github.com/opensearch-project/dashboards-notifications/pull/257/files?diff=split&w=0#r1747503523

**Review by AWSHurneyt — APPROVED** [review]
Changes LGTM. Approving assuming the CI passes, and other comments are addressed.

File: public/utils/constants.ts [hunk 4]
```diff
@@ -51,7 +55,11 @@ export const BREADCRUMBS = Object.freeze({
     text: 'Create recipient group',
     href: `#${ROUTES.CREATE_RECIPIENT_GROUP}`,
   },
-  EDIT_RECIPIENT_GROUP: { text: 'Edit recipient group' },
+  EDIT_RECIPIENT_GROUP: { text: 'Edit recipient group', href: `#${ROUTES.EDIT_RECIPIENT_GROUP}`},
+  EDIT_RECIPIENT_GROUP_DETAILS: (name: string) => ({
+    text: `${name}`,
```
Comment (amsiglan) [inline public/utils/constants.ts:60]: Can be simplified to name, don't need the string literal

File: public/pages/Emails/CreateRecipientGroup.tsx (hunk 2 above)
Comment (amsiglan) [reply public/pages/Emails/CreateRecipientGroup.tsx:78]: Aah, I see what you mean, ignore my comment.

File: public/utils/constants.ts (hunk 4 above)
Comment (amsiglan) [reply public/utils/constants.ts:60]: Same in other places

Comment (riysaxen-amzn) [reply public/utils/constants.ts]: addressed

**Review by amsiglan — APPROVED** [review]

**Pull request merged**